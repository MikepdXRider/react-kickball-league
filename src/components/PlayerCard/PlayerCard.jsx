import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

export default function PlayerCard({playerDataObj}) {
    return (
        <Link to={`/player/${playerDataObj.id}`}>
            <article>
                <h3>{playerDataObj.name}</h3>
                <h6>{playerDataObj.created_at}</h6>
                <h5>Position: {playerDataObj.positoin}</h5>
            </article>
        </Link>    
    )
}

PlayerCard.propType = {
    playerDataObj: PropTypes.shape({
        created_at: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        team_id: PropTypes.number.isRequired
    })
}