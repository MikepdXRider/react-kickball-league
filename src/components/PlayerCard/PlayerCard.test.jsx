import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import PlayerCard from "./PlayerCard.jsx";


const playerDataObj = {
    created_at: 'test-utc',
    id: 1,
    name: 'test-name',
    team_id: 3
}

it('renders PlayerCard component', () => {
    const { container } = render(<PlayerCard playerDataObj={playerDataObj} /> )
    const playerName = screen.getByText(playerDataObj.name);

    expect(playerName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})