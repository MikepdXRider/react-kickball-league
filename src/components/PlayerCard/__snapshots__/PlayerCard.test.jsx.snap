// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayerCard component 1`] = `
<div>
  <article>
    <h3>
      test-name
    </h3>
    <h6>
      Date Created: 
      Invalid Date
    </h6>
    <h5>
      Position: 
    </h5>
  </article>
</div>
`;
